File: app/src/main/java/telstra/sohan/com/sohantelstracodingtest/Presenter/GetDataCallBack.java
package telstra.sohan.com.sohantelstracodingtest.Presenter;

import android.content.Context;

import java.util.List;

import telstra.sohan.com.sohantelstracodingtest.Model.RowsDataDto;


public interface GetDataCallBack {
    interface View{
        void onGetDataSuccess(String message, List<RowsDataDto> list,String title);
        void onGetDataFailure(String message);
    }
    interface Presenter{
        void getDataFromURL(Context context, String url);
    }
    interface Interactor{
        void initRetrofitCall(Context context, String url);

    }
    interface onGetDataListener{
        void onSuccess(String message, List<RowsDataDto> list, String title);
        void onFailure(String message);
    }
}

File: app/src/main/java/telstra/sohan/com/sohantelstracodingtest/presenter/GetDataCallBack.kt
package telstra.sohan.com.sohantelstracodingtest.presenter

import android.content.Context
import telstra.sohan.com.sohantelstracodingtest.model.FactsDataDto


interface GetDataCallBack {
    interface View {
        fun onGetDataSuccess(listFactsDataDto: List<FactsDataDto>, title: String)
        fun onGetDataFailure(message: String)
    }

    interface Presenter {
        fun getCall(context: Context)
    }

    interface Interact {
        fun initRetrofitCall(context: Context)
    }

    interface OnGetDataListener {
        fun onSuccess(listFactsDataDto: List<FactsDataDto>, title: String)
        fun onFailure(message: String)
    }
}
